### ISSUE TITLE
Alternate behavior in test vs. development

### ISSUE THREAD

**Issue by jejacks0n** [rails/sprockets-rails#115]
Ok, so in fairness this isn't an issue, because I couldn't for the life of me figure out why, but please read on because I'm still looking for help.

I wrote [Teaspoon](https://github.com/modeset/teaspoon), to help with javascript testing in rails. I've tried to do a bunch of things to make this nicer, and one of those things was to provide coverage reports.

I'm using istanbul to generate javascript files with instrumentation. You can see where I apply [the logic to run the requested asset through istanbul](https://github.com/modeset/teaspoon/blob/0.8/lib/teaspoon/instrumentation.rb#L61) to get the instrumentation.

It's worth noting that I was unable to use a postprocessor for this because the instrumented files should never be cached -- though my implementation does take advantage of the asset caching because it comes after the response is generated from sprockets.

I override the "call" method in Sprockets::Environment and inject that in [the engine](https://github.com/modeset/teaspoon/blob/0.8/lib/teaspoon/engine.rb#L32)

To summarize, we hook into sprockets so we can run the asset through a postprocessor of sorts so any request to a javascript asset with the "instrument=1" query param will get processed through istanbul.

Now, this works absolutely fine, and has since ~3.2.10 (around the time I started teaspoon). Though, in my integration specs (using capybara/rails), I cannot for the life of me get the same behavior. It used to work, but something has changed with Journey and how sprockets adds to the middleware.. any ideas?

[ note: I'm creating and closing the issue, but this was the best way for me to reach out and help get some clarification ]
